File: injector/src/test/java/net/mready/inject/FieldInjectionTest.java
package net.mready.inject;

import org.junit.Test;

import javax.inject.Inject;

import static org.junit.Assert.assertNotNull;

public class FieldInjectionTest {
    @Test
    public void fieldsInjected() {
        Injector injector = Injector.with();
        Target target = new Target();
        injector.injectFields(target);
        assertNotNull(target.a);
    }


    public static class Target {
        @Inject
        private A a;
    }

    public static class A {

    }
}

File: injector/src/test/kotlin/net/mready/inject/FieldInjectionTest.kt
package net.mready.inject

import org.junit.Assert.assertNotNull
import org.junit.Test
import javax.inject.Inject

class FieldInjectionTest {
    @Test
    fun fieldsInjected() {
        val injector = Injector.with()
        val target = Target()
        injector.injectFields(target)
        assertNotNull(target.a)
    }

    class Target {
        @Inject lateinit var a: A
    }

    class A
}
